include/deal.II/numerics/vector_tools_boundary.templates.h

```diff
@@ -283,8 +283,9 @@ namespace VectorTools
                       // dofs on this face
                       face_dofs.resize(fe.n_dofs_per_face(face_no));
                       face->get_dof_indices(face_dofs, cell->active_fe_index());
-                      const std::vector<Point<spacedim>> &dof_locations =
+                      std::vector<Point<spacedim>> dof_locations =
                         fe_values.get_quadrature_points();
+                      dof_locations.resize(fe.n_dofs_per_face(face_no));
```

Comment:
This implies a memory allocation on every such face on the boundary. We probably have more expensive operations in there, so I think we should get this in like this if we do not see a simple alternative solution. But we might at some point want to look at this again.